### PR TITLE
Add icon

### PR DESCRIPTION
**Pull request by samrocksc**
#### What's this PR do?

It implements `<Icon/>` as a component.  This will be useful in future components, it also takes the hassle out of creation of an icon.  It ships with 2 modes:

`<Icon>search</Icon>` and `<Icon toggle>search</Icon>`.  This helps create togglable icons
#### What are the important parts of the code?

index.js to tie the functionality from icon.js
#### How should this be tested by the reviewer?

pull it->npm run build->npm start->browser: localhost:8080/example
#### Is any other information necessary to understand this?

Not really it's cut and dry
#### What are the relevant tickets? (Please add `closes`, `refs`, etc)

none

### CODE REVIEW

File: example/index.js [hunk 1]
```diff
@@ -24,6 +24,11 @@ const exampleTree = (
       <Checkbox label='With Id' id='checkboxId' />
       <Checkbox label='Without Id' />
 
+    <h4>Icons</h4>
+      <Icon>star_rate</Icon> Regular Icon
```
Comment (phated) [inline example/index.js]: why did you choose a child string instead of a property on the component?

**Comment by samrocksc** [comment]
We could do it that way with the option of the child as the label for it.  What do you think is the best guidance for this? 

mdl isn't really specific on icons; however, it uses them frequently as spans in documentation.

**Comment by samrocksc** [comment]
Long conversation short, it's kind of your guidance.  I was under the impression that we wanted to keep this as child based as possible.  I think this is the best way to signify an icon as you can add a className addition to the <span> if needed, or even add a label function.

**Comment by samrocksc** [comment]
Fixed the syntax to your liking

File: src/icon.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+const baseClasses = {
+  'react-material-design-lite-icons': true
+};
+
+class Icon extends React.Component {
+
+  render(){
+    const {
+      children,
```
Comment (phated) [inline src/icon.js]: unused

File: src/icon.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        {name}
+      </span>
+    );
+  }
+}
+
+Icon.propTypes = {
+  className: React.PropTypes.string,
+  toggle: React.PropTypes.bool,
+  name: React.PropTypes.string
```
Comment (phated) [inline src/icon.js]: name is required

**Comment by samrocksc** [comment]
llintd, should be fine

**Pull request merged**